libraries/mbed/targets/hal/TARGET_NXP/TARGET_LPC81X/device.h

```diff
@@ -29,7 +29,7 @@
 #define DEVICE_SERIAL_FC        1
 
 #define DEVICE_I2C              1
-#define DEVICE_I2CSLAVE         0
+#define DEVICE_I2CSLAVE         1
```

Comment:
Hi Martin,

As explained in the comments for the pull request, the I2C slave can be
enabled and the slave block read/write operations are fine. The additional
work in I2CSlave.cpp is needed for a correct operation of the slave byte
read/write operations. This change includes platforms such as the LPC812 and
LPC824 because they have different I2C engines for master and slave.

Again, the currently accepted LPC824 I2C slave lib has that same problem and
doesn't work correctly for the slave byte read/writes.

 

Wim

  _____  

From: Martin Kojtal [mailto:notifications@github.com] 
Sent: donderdag 12 februari 2015 9:32
To: mbedmicro/mbed
Cc: Wim
Subject: Re: [mbed] Create pwmout_api.c (#906)

 

In libraries/mbed/targets/hal/TARGET_NXP/TARGET_LPC81X/device.h
<https://github.com/mbedmicro/mbed/pull/906#discussion_r24565131> :

> @@ -29,7 +29,7 @@
>  #define DEVICE_SERIAL_FC        1
>  
>  #define DEVICE_I2C              1
> -#define DEVICE_I2CSLAVE         0
> +#define DEVICE_I2CSLAVE         1

Isn't this the target which will need modifications to have slave enabled?

-
Reply to this email directly or view it on
<https://github.com/mbedmicro/mbed/pull/906/files#r24565131>  GitHub.
<https://github.com/notifications/beacon/AKMUNJ5mrGex1x6ADfFng6gpyjnPdGHRks5
nrFyBgaJpZM4DfF7f.gif>